### ISSUE TITLE
useStoreState doesn't react to changes in the state after Fast Refresh in React Native

### ISSUE THREAD

**Issue by shancial** [ctrlplusb/easy-peasy#477]
Hi,

I added easy-peasy(v.3.3.0) successfully into a new React Native(v0.62.2) project but noticed that  a function component using useStoreState will not react to the state after editing the component code. My button will send a request, receive a new picture of a dog and set that dog into the state but my component will not detect the state change. I have React Native's Fast Reload enabled.

Here is my test component (TestComponent.tsx):
````
import ...

export function TestComponent() {
  const dog = useStoreState((state) => state.test.dog);
  const getDog = useStoreActions((actions) => actions.test.getDog);

  const handleGetDogPress = useCallback(async () => {
    try {
      await getDog();
    } catch {
      console.log('Simulated api error occurred');
    }
  }, [getDog]);

  return (
    <View style={styles.container}>
      {dog && <Image style={styles.logo} source={{uri: dog}} />}
      <Button title="Get Dog" onPress={handleGetDogPress} />
    </View>
  );
}

const styles = StyleSheet.create(...)
````

[Fast Refresh documentation](https://reactnative.dev/docs/fast-refresh) points out that it'll try to preserve the state if a change is done to a file that only exports a React component:
> When possible, Fast Refresh attempts to preserve the state of your component between edits. In particular, useState and useRef preserve their previous values as long as you don't change their arguments or the order of the Hook calls.

If I change the order of the useStoreState and useStoreActions, the useStoreState hook will work until I make a modification that makes Fast Refresh try to preserve values of useStates and useRefs. Putting `// @refresh reset` somewhere in the file also works but it forces the component to re-render on every save. Disabling Fast Refresh works as well but it makes the developer experience really bad.

Maybe the bug is related to the useRef hooks used in the useStoreState? I checked the code but I'm afraid I don't know enough about hooks/redux/easy-peasy yet to fix it myself.

Is there anyone that can look into this?

**Comment by ctrlplusb** [comment]
Hi @shancial 

Thanks for this report. I can empathise with that pain. I'll def make some time to look into this for you.

**Comment by shancial** [comment]
Thanks!

I can provide you with the test project code that I tested this with, but I basically used [the official guide](https://reactnative.dev/docs/environment-setup) (`npx react-native init AwesomeTSProject --template react-native-template-typescript`) to set up a base and added easy-peasy with a property, setter and a thunk.

**Comment by Nr9** [comment]
same issue.

**Comment by shancial** [comment]
I seems like Fast Refresh / react-refresh [is also coming to create-react-app](https://github.com/facebook/create-react-app/pull/8582) in the future.

**Comment by DaneEveritt** [comment]
Can also confirm this is happening to me as well. Anything I can do that would help with getting this looked into? :)

**Comment by ctrlplusb** [comment]
Hi all, I want to look into this. Are you all running the Expo CLI, and RN Web?

**Comment by DaneEveritt** [comment]
I am using the standard react-native CLI and React Native.

**Comment by hutber** [comment]
I confirm the same as @rfermann issue is happening with nextjs.

**Comment by mrapczynski** [comment]
I just ran into this too building an app that uses Fast Refresh, but for the web, and not with React Native.

**Comment by jmyrland** [comment]
We're also starting to see the issue of fast refresh not working properly with easy-peasy. I've reproduced the issue in [this codesandbox](https://codesandbox.io/s/solitary-framework-p6ujq?file=/src/Counter.tsx):

![chrome_NVK65pZfLX](https://user-images.githubusercontent.com/837651/149137897-f0eadd3f-42cc-49de-a596-3cab397cd501.gif)

Any ideas as to what is causing this @ctrlplusb ?

**Comment by ctrlplusb** [comment]
Fixed in v5.1.0 🎉
